lib/sinatra/base.rb

```diff
@@ -1151,11 +1152,17 @@ def static!(options = {})
 
       path = File.expand_path(path)
       return unless path.start_with?("#{File.expand_path(public_dir)}/")
-
       return unless File.file?(path)
 
       env['sinatra.static_file'] = path
       cache_control(*settings.static_cache_control) if settings.static_cache_control?
+
+      if settings.respond_to?(:static_headers) && settings.static_headers
+        settings.static_headers.each do |k, v|
+          headers[k] = v
+        end
+      end
```

Comment:
Actually, there's no need to loop

```suggestion
      headers(settings.static_headers) if settings.static_headers?
```

Reply:
Should we add this remove before or after `cache_control(*settings.static_cache_control) if settings.static_cache_control?`?

If we add it before, `static_cache_control` have precedence, if we add it after, you can overwrite headers like `Cache-Control`, with `static_headers`. What do we want?

Reply:
Good question, personally I think that calling ` headers(settings.static_headers) ` after `Cache-Control` makes more sense, since it gives the users more control. It feels more explicit and flexible that way, at least to me.

Reply:
Sounds reasonable to me